oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/h2/dao/H2TraceQueryDAO.java

```diff
@@ -82,15 +77,36 @@ public TraceBrief queryBasicTraces(long startSecondTB,
                                                        .provider()
                                                        .getService(ConfigService.class);
             searchableTagKeys = Arrays.asList(configService.getSearchableTracesTags().split(Const.COMMA));
-            if (searchableTagKeys.size() > maxSizeOfArrayColumn) {
-                this.searchableTagKeys = searchableTagKeys.subList(0, maxSizeOfArrayColumn);
-            }
         }
 
         StringBuilder sql = new StringBuilder();
         List<Object> parameters = new ArrayList<>(10);
 
-        sql.append("from ").append(SegmentRecord.INDEX_NAME).append(" where ");
+        sql.append("from ").append(SegmentRecord.INDEX_NAME);
+
+        /**
+         * If no tags condition, only query segment table, the SQL should be:
```

Comment:
It seems there is only this one place having comments to describe the SQL building logic. Would you add a reference to other query methods? It would be hard to know that there is a good explanation here when reading alarm/event/log query.

Or, is it better to add this as a general comment to the annotation? It should be easier to find and refer?